content/guides/full-text-search.md

```diff
@@ -0,0 +1,215 @@
+author: vkarpov15
+enableTableOfContents: true
+createdAt: '2024-09-17T13:24:36.612Z'
+updatedOn: '2024-09-17T13:24:36.612Z'
+---
+
+
+The `tsvector` type enables you to use full text search on your text content in Postgres. Full text search allows you to search text content in a more flexible way than using `LIKE`: full text search also supports features like _stemming_, which means searching for the word "run" will match variations like "ran" and "running".
+
+## Steps
+
+* Set up a table with a `tsvector` column
+* Execute your first full text search
+* Search for multiple words
+* Rank the results
+* Create a GIN index
+
+## Set up a table with a `tsvector` column
+
+To set up full text search, you need to create a column of type `tsvector` that will enable full text search. You can run the following `CREATE TABLE` statement in the Neon SQL Editor or from a client such as `psql` that is connected to Neon. This statement will create a table with a column `searchable` of type `tsvector`.
+
+```sql
+CREATE TABLE documents (
+  id SERIAL PRIMARY KEY,
+  title TEXT,
+  body TEXT,
+  searchable tsvector
+);
+```
+
+Next, insert two new rows into the `documents` table. The [`to_tsvector()` function](https://www.postgresql.org/docs/current/textsearch-controls.html) takes in a language and the text content to tokenize. In the following example, the text content is the `title` and `body` columns concatenated together.
+
+```sql
+INSERT INTO documents (title, body, searchable)
+  VALUES (
+    'PostgreSQL Full-Text Search',
+    'This is an introduction to full-text search in PostgreSQL.',
+    to_tsvector('english', 'PostgreSQL Full-Text Search This is an introduction to full-text search in PostgreSQL.')
+  );
+
+INSERT INTO documents (title, body, searchable)
+  VALUES (
+    'My Mashed Potatoes Recipe',
+    'These amazing homemade mashed potatoes are perfectly rich and creamy, full of great flavor, easy to make, and always a crowd fave!',
+    to_tsvector('english', 'My Mashed Potatoes Recipe These amazing homemade mashed potatoes are perfectly rich and creamy, full of great flavor, easy to make, and always a crowd fave!')
+  );
+```
+
+Once you have inserted the new rows, try running `SELECT * FROM documents;`. You will see that the data stored in the `searchable` column looks like the following.
+
+```
+'full':3,12 'full-text':2,11 'introduct':9 'postgresql':1,16 'search':5,14 'text':4,13
+```
+
+Internally, `to_tsvector()` uses a parser to break the text content into tokens for easier searching.
+
+## Execute your first full text search
+
+You can execute a full text search query using a `WHERE` clause with `@@ to_tsquery('english', 'content here')` as follows. The following query will return the "mashed potatoes" row because, although the word "flavorful" does not appear in that row's text content, the word "flavor" does. And "flavor" matches "flavorful".
```

Comment:
```suggestion
You can execute a full text search query using a `WHERE` clause with `@@ to_tsquery('english', 'content here')` as shown below. The following query returns the "mashed potatoes" row because, although the word "flavorful" does not appear in that row's text content, the word "flavor" does. And "flavor" matches "flavorful".
```